core/src/main/java/org/apache/druid/math/expr/Parser.java

```diff
@@ -171,13 +171,9 @@ public static Expr applyUnappliedIdentifiers(Expr expr, Expr.BindingDetails bind
       return expr;
     }
     List<String> unapplied = toApply.stream()
-                                     .filter(x -> bindingDetails.getFreeVariables().contains(x))
+                                     .filter(x -> bindingDetails.getRequiredColumnsList().contains(x))
```

Comment:
Suggest using `getRequiredColumns` here, since `contains` should be cheaper on the Set vs ArrayList if there are lots of columns (and skip the ArrayList creation too)

Reply:
Oops, `getRequiredColumns` originally returned a list, but I changed it to `getRequiredColumnsList` and added `getRequiredColumns` to return a set for this reason but forgot to make sure I updated all the places.